subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/DefaultTypeMetadataStore.java

```diff
@@ -158,9 +158,10 @@ private static String calculateDisplayName(Iterable<? extends PropertyAnnotation
                         problem.withId(ValidationProblemId.ANNOTATION_INVALID_IN_CONTEXT)
                             .forProperty(propertyAnnotationMetadata.getPropertyName())
                             .reportAs(ERROR)
-                            .withDescription(() -> String.format("has invalid annotation @%s", annotationType.getSimpleName()))
-                            .happensBecause(() -> "The '@" + propertyType.getSimpleName() + "' annotation cannot be used in this context")
```

Comment:
Did we actually print the wrong type here in the past?

Reply:
Yes, we did. Though we never ended up in this branch, since it we always allowed all modifiers.